Mastercard.Developer.OAuth1Signer.RestSharpV2/Signers/RestSharpSigner.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using Mastercard.Developer.OAuth1Signer.Core;
+using Mastercard.Developer.OAuth1Signer.Core.Signers;
+using RestSharp;
+
+#pragma warning disable 1591
+
+namespace Mastercard.Developer.OAuth1Signer.RestSharpV2.Signers
+{
+    /// <summary>
+    /// Utility class for signing RestSharp request objects.
+    /// </summary>
+    public sealed class RestSharpSigner : BaseSigner
+    {
+        public RestSharpSigner(string consumerKey, RSA signingKey) : base(consumerKey, signingKey)
+        {
+        }
+
+        public RestSharpSigner(string consumerKey, RSA signingKey, Encoding encoding) : base(consumerKey, signingKey, encoding)
+        {
+        }
+
+        public void Sign(Uri baseUri, IRestRequest request)
+        {
+            if (baseUri == null) throw new ArgumentNullException(nameof(baseUri));
+            if (request == null) throw new ArgumentNullException(nameof(request));
+
+            // Build the full request URI
+            var fullUri = new StringBuilder();
+            fullUri.Append(baseUri.ToString().TrimEnd('/'));
+            var resource = request.Resource;
+            if (!string.IsNullOrEmpty(resource))
+            {
+                resource = resource.TrimStart('.');
+                resource = resource.TrimStart('/');
+                fullUri.Append("/").Append(resource);
+            }
+
+            // Add query params
+            var parameterString = new StringBuilder();
+            foreach (var requestParameter in request.Parameters.Where(p => p.Type == ParameterType.QueryString))
+            {
+                parameterString
+                    .Append(parameterString.Length > 0 ? "&" : string.Empty)
+                    .Append(Uri.EscapeDataString(requestParameter.Name))
```

Comment:
We have two possible null assignment to a non-nullable entity here. I'm not sure if `requestParameter.Name` and `requestParameter.Value` can happen to be null, but we could add a quick test to ignore such parameters.